### DISCUSSION TITLE
Oracle query aggregation generates "AS" for alias table name

### DISCUSSION THREAD

**Discussion by mborgues** [TriPSs/nestjs-query#322] [General]
"AS" for table name is not supported in Oracle, is it generated by nest-query or TypeORM?
Is there any solution for this?

![image](https://github.com/user-attachments/assets/a2a80813-7015-4ec3-8c6b-c8fbcac384d7)

Without "AS"
![image](https://github.com/user-attachments/assets/f1162e6a-a4cd-4d99-bc9b-f0fb5859a5ea)

**Comment by TriPSs** [comment]
I think this is something TypeORM does but i'm not 100% sure. Looking at the outputted query I'm also wondering what version of the lib you are running? The `__nestjsQuery__entityIndex__` should almost never be used anymore I think as it was removed for the majority of the queries because of performance drawbacks.

**Reply by mborgues** [reply]
I believe it is indeed related to typeORM, after researching a little more I was able to create a patch to adjust the typeORM queries, for now I was able to solve this situation, thank you very much.

About the version I am using:
"@ptc-org/nestjs-query-core": "^7.1.0",
"@ptc-org/nestjs-query-graphql": "^7.1.0",
"@ptc-org/nestjs-query-typeorm": "^7.1.0",
"typeorm": "^0.3.20"
"@nestjs/typeorm": "^10.0.2"

Patch to add to my app to adjust the query in TypeORM

```
import { SelectQueryBuilder } from 'typeorm';

const originalGetQuery = SelectQueryBuilder.prototype.getQuery;

SelectQueryBuilder.prototype.getQuery = function () {
  // Chama o método original para obter a consulta
  const query = originalGetQuery.call(this);

  // Remove "AS" de aliases de tabelas e subconsultas
  const patchedQuery = query.replace(/(\sAS\s)+/gi, ' ');

  return patchedQuery;
};
```

**Reply by mborgues** [reply]
I believe I have found a new inconsistency in the generated group by condition, Oracle does not accept the use of aliases in this clause, the table name and field name must be used, the use of the alias is only allowed in Order By, here is a generated example and the correct form:

incorrect:
```sql
SELECT "InformacoesGeraisEntity"."CODIGOUF" "GROUP_BY_codigoUf",
       COUNT("InformacoesGeraisEntity"."NUMERORAINFGER") "COUNT_numeroRa"
  FROM "ACD_INFORMACOESGERAIS" "InformacoesGeraisEntity"
 GROUP BY "GROUP_BY_codigoUf"
 ORDER BY "GROUP_BY_codigoUf" ASC
``` 
 correct:
```sql
SELECT "InformacoesGeraisEntity"."CODIGOUF" "GROUP_BY_codigoUf",
       COUNT("InformacoesGeraisEntity"."NUMERORAINFGER") "COUNT_numeroRa"
  FROM "ACD_INFORMACOESGERAIS" "InformacoesGeraisEntity"
 GROUP BY "InformacoesGeraisEntity"."CODIGOUF"
 ORDER BY "GROUP_BY_codigoUf" ASC
```

**Reply by TriPSs** [reply]
This starts to sound like the same issues as in #159, could that be correct?

**Reply by mborgues** [reply]
Yes, it's the same problem, I created a patch to adjust typeOrm queries at runtime, for now it's working, but I don't know for how long, or what problem it might cause me.

**Reply by TriPSs** [reply]
Could you test [this comment](https://github.com/TriPSs/nestjs-query/issues/159#issuecomment-1900631667)?

**Reply by mborgues** [reply]
Doesn't worked:

![image](https://github.com/user-attachments/assets/e3738e4c-87b8-4621-b474-609554fbcaed)

**Reply by TriPSs** [reply]
Aah yea, just now I see that that flag only affects the group by of dates, sorry! Think for this to work a custom implementation for Oracle needs to be made in `filter-query.builder.ts:242`:
```ts
return aggregatedGroupBy.reduce((prevQb, aggregatedField) => {
      return prevQb.addGroupBy(prevQb.escape(AggregateBuilder.getGroupByAlias(aggregatedField.field)))
}, qb)
```

```ts
return aggregatedGroupBy.reduce((prevQb, aggregatedField) => {
      return prevQb.addGroupBy(prevQb.escape(aggregatedField.field)))
}, qb)
```

You could try something like this and if it works create a patch for it inside your repo.

**Reply by mborgues** [reply]
Thanks for the help, I will need to study your framework a little more to make the adjustment, because with the change in the way you mentioned it uses the name of the property and not the name of the column in the database, I still need to generate the field as in the select, for example group by "ModeloVeiculoTerceiroEntity"."CODIGOMODELOVEICTER" and not group by "codigoVeiculoTerceiro", but for now I have a working solution until I managed to do this.

**Reply by mborgues** [reply]
Sorry, now that I saw that the function parameters already have the necessary information, here is the function that adjusts without any problems:

```js
applyAggregateGroupBy(qb, aggregatedGroupBy, alias) {
    if (!aggregatedGroupBy) {
        return qb;
    }
    return aggregatedGroupBy.reduce((prevQb, aggregatedField) => {
        const column = `${alias}.${aggregatedField.field}`;
        return prevQb.addGroupBy(column);
    }, qb);
}
```
